testsuite/mockserver/classes.py

```diff
@@ -37,24 +37,39 @@ class SslCertInfo:
 
 @dataclasses.dataclass(frozen=True)
 class MockserverInfo:
-    host: str
-    port: int
     base_url: str
-    ssl: typing.Optional[SslCertInfo]
 
     def url(self, path: str) -> str:
         """Concats ``base_url`` and provided ``path``."""
         return url_util.join(self.base_url, path)
 
+    def get_host_header(self) -> str:
+        raise NotImplementedError
+
+
+@dataclasses.dataclass(frozen=True)
+class MockserverTcpInfo(MockserverInfo):
+    host: str
+    port: int
+    ssl: typing.Optional[SslCertInfo]
+
     def get_host_header(self) -> str:
         if self.port == 80:
             return self.host
         return f'{self.host}:{self.port}'
 
 
-class MockserverSslInfo(MockserverInfo):
+class MockserverSslInfo(MockserverTcpInfo):
```

Comment:
how would you differ MockserverTcpInfo from MockserverSslInfo, optional ssl attribute was there for purpuse